app-frontend/src/app/core/services/map.service.js

```diff
@@ -379,6 +379,16 @@ class MapWrapper {
         this.heldState = state;
         return this;
     }
+
+    /** Get the extent from a project (if it exists) and fit the map to it
+      * @param {Project} project to hold, e.g. specifying latlng center and zoom
+      * @returns {this} this
+      */
+    fitProjectExtent(project) {
```

Comment:
Similar to the other PR - I'd rather have this in the controller which supplies the project than in the map service, since the map shouldn't have any concept of what a project is. I'd like to follow the same pattern as we use elsewhere when setting map view state.

If this is a function we use frequently, then it may make sense to create a separate MapUtils service for utility functions.
Something like
```javascript
class MapUtils {
  def extendToBounds(extent) {
    return L.geoJSON(extent).getBounds();
  }
}
```
```javascript
if (mapWrapper && project && project.extent) {
  mapWrapper.map.fitBounds(MapUtils.extentToBounds(project.extent));
}
```